### DISCUSSION TITLE
Could not increase the Playwright default timeout limit - 30000ms through artillery tests

### DISCUSSION THREAD

**Discussion by sruthii38** [artilleryio/artillery#1939] [Artillery]
I have a scenario where an action to upload 3k records through xls is taking around 30s and sometimes more. We are planning to increase this limit of xls record to 10k and I cannot validate the performance because playwright's default time is timing out 50% of my transactions when I try it for 3k limit. I have tried the following.

Action timeout: .click{timeout:60000} - no impact Timeout happened at 3000ms

test.setTimeout(60000) - worked only for Playwright and errored when Artillery picked with error - test.setTimeout can only be used with test.

contextOptions:
 defaultTimeout: 60000 - no impact timeouts happened at 30000ms

**Comment by hassy** [comment]
hi @sruthii38 👋 

this should work:

```
config:
  engines:
    playwright:
      defaultTimeout: 60000
```

please make sure you're on the most recent version of the engine as well (v0.2.2)

**Reply by sruthii38** [reply]
Hi hassy, the version you mentioned does not align with either my artillery or Playwright. here is what I have
Artillery -  2.0.0-31
Playwright - 1.33.0
they both are ahead of v0.2.2, can you please clarify if I am not on right track. Thanks

**Reply by sruthii38** [reply]
just as benefit of doubt, I got the latest v for artillery 2.0.0-32. But that did not solve the issue, the timeouts are still happening at 30000ms. Please someone reply if we have a fix, this is such a blocker for my tests that can't even get baseline for existing functionality.

**Reply by hassy** [reply]
@sruthii38 you need to update Artillery/Playwright integration with:

```
npm install artillery-engine-playwright@latest
```

(Depending on how your project is set up it may need to be a global install)

We will bundle the integration into Artillery itself in the upcoming release, but right now it needs to be updated separately.

**Reply by jracabado** [reply]
For anyone reading this, the `defaultTimeout` is in seconds and not in milliseconds as @hassy example shows.

**Comment by sruthii38** [comment]
I see this warning 
⠙ WARNING: engine contextOptions specified but module artillery-engine-contextOptions could not be loaded
Error: Cannot find module 'artillery-engine-contextOptions'

**Reply by hassy** [reply]
Your config isn't right. Try using the configuration I shared above making sure that `defaultTimeout` is under `playwright` which is under `engines` in `config`. That should work.

**Comment by sruthii38** [comment]
Okay, I took off the contextOptions and gave it exactly how you quoted. But I still see timeout at 30000ms
config:
   target: 'xxx'
   environments:
     beta:
       target:'xxx'
     gamma:
       target:'xxx'
   engines:
     playwright:
       defaultTimeout: 60000
![image](https://github.com/artilleryio/artillery/assets/135713421/96677a93-b7db-4de1-bdf9-ff3deef4590a)
for some reason, I could not correct the hierarchy so posting a pic

**Reply by sruthii38** [reply]
@hassy please... help me what is wrong with my config that default timeout is not changing from 30000ms. Your time is appreciated!

**Reply by hassy** [reply]
can you confirm you've installed the most recent version of `artillery-engine-playwright`?

**Reply by sruthii38** [reply]
yes sir, 
![image](https://github.com/artilleryio/artillery/assets/135713421/0b267e79-9357-4d8b-9a48-e40d5b76f08a)
playwright
![image](https://github.com/artilleryio/artillery/assets/135713421/b397fb99-4c30-4569-b38a-1245c75e726b)

and I am not getting the any warning regarding config too but timeout is still at 30000ms

**Reply by sruthii38** [reply]
@hassy not sure you got my reply above so @mentioning for attention. Please let me know why it wouldn't work. Thanks

**Reply by sruthii38** [reply]
@hassy Can you please help me with this, it has been a blocker for my delivery.

**Reply by hassy** [reply]
@sruthii38 try the new release of Artillery - v2.0.0-33. It bundles the Playwright integration. I suspect you're not running the latest version of `artillery-engine-playwright` currently which is why the timeout config isn't working for you.

**Reply by sruthii38** [reply]
@hassy I just pulled the latest for Artillery yesterday which was v2.0.0-32. I pulled the latest today aswell for 33 and ran my tests. I am not seeing timeouts now but the tests are not stopping at the specified time. I was running 3m 30s tests and it was not stopping even after 10mins of wait. I commented out the "defaultTimeout" and it is stopping at the right time. Do you think I need to make any changes to stop it at the time it is supposed to?

PS: artillery-engine-playwright has been on 0.2.2 on v2.0.0-32 and remained the same after todays upgrade.
![image](https://github.com/artilleryio/artillery/assets/135713421/f6c107c7-5dea-411b-a065-42339ea382c3)

please advice, thank you

**Reply by sruthii38** [reply]
@hassy adding to the update - when defaultTimeout is used, it is basically not giving any metrics both on logs or reports and just spinning as if execution is in progress.

**Reply by sruthii38** [reply]
@hassy without aggregateByName, I am not getting the aggregate metrics I need for my endpoint. But, passing aggregateByName with defaultTimeout is not working - meaning not getting timeouts and the tests never stop either. The tests don't stop even if the default Timeout is set to 30000ms like below.
![image](https://github.com/artilleryio/artillery/assets/135713421/1477dbb8-429a-4c9e-9570-2a0a01b46714)

tests does not stop with just defaultTimeout too.  
![image](https://github.com/artilleryio/artillery/assets/135713421/ac6136b7-8b33-41bd-9d23-e4f564012871)

they only stop if I give aggregateByName like below but they are causing timeouts
![image](https://github.com/artilleryio/artillery/assets/135713421/c858c98b-fbc4-4dca-81dc-5f42d60821cc)

So, I think this a issue with defaultTimeout or I don't know how to use it. Please look into this to unblock the test executions(I hope this helps other folks who are looking to use this option)

Thank you

**Reply by hassy** [reply]
The value of `defaultTimeout` is expected to be in seconds, so you need to set it to just 60.

I can't reproduce `aggregateByName` not working when used with `defaultTimeout`. They seem to work as expected for me. The issue is likely to be with your test scripts or your environment.

**Reply by sruthii38** [reply]
There you go, that is the issue I was always using 60000 for defaultTimeout to bump it from default. Not sure if there is no polling in between, and looks like it is hanging at that high timeouts. Everything worked once after I used 60s. Thanks @hassy for being very responsive. Have a good rest of your day!

**Reply by hassy** [reply]
Glad to hear it's working! :)

**Reply by Romarionijim** [reply]
@sruthii38 @hassy The globalTimeout in playwright.config.ts does not have an affect?  meaning I need to set the timeout explicitly in each yml test file?

**Reply by hassy** [reply]
@Romarionijim Artillery does not currently load the `playwright.config.ts` file. It's used by the Playwright Test Runner which Artillery does not support yet. Artillery's integration uses Playwright as a library.
